core/arch/arm/include/mm/core_mmu.h

```diff
@@ -115,6 +116,13 @@ struct core_mmu_phys_mem {
 		__used __section("phys_mem_map_section") = \
 		{ #addr, (type), (addr), (size) }
 
+#define __register_sdp_mem2(pa, sz, id) \
+	static const struct core_mmu_phys_mem __phys_sdp_mem_ ## id \
+		__used __section("phys_sdp_mem_section") = \
+		{ .type = MEM_AREA_SDP_MEM, .addr = (pa), .size = (sz), }
+
+#define __register_sdp_mem1(pa, sz, id)	__register_sdp_mem2(pa, sz, id)
+#define register_sdp_mem(pa, sz)	__register_sdp_mem1(pa, sz, __LINE__)
```

Comment:
`__LINE__` won't work if `register_sdp_mem()` is used twice on the same line (in a multi-line macro, for instance). You should use `__COUNTER__` instead.